### PR TITLE
feat: refactoring game service

### PR DESCRIPTION
**Pull request by hrubalski**

### CODE REVIEW

File: lib/big_apple_game.dart [hunk 1]
```diff
@@ -166,17 +162,17 @@ class BigAppleGame extends CommonGame with ScaleDetector {
   void _startSaveTimer() {
     _saveGameTimer?.cancel();
     _saveGameTimer = Timer.periodic(AppDuration.saveGameDuration, (timer) {
-      gameBloc.add(const GameSaveEvent());
+      // gameBloc.add(const GameSaveEvent());
```
Comment (Sasha071201) [inline lib/big_apple_game.dart]: why?

File: lib/common/components/building_component.dart [hunk 2]
```diff
@@ -1,43 +1,40 @@
...
     super.key,
     required this.building,
-    required super.size,
     super.anchor = Anchor.center,
     bool markAsBuild = false,
   }) : super(
-          priority: building.coordinates.y.toInt() +
-              (building.building.type == ManufactoryType.coalElectricStation ? 100 : 0),
-          position: Vector2(building.coordinates.x, building.coordinates.y),
+          priority: building.y.toInt() + (building.type == BuildingType.coalElectricStation ? 100 : 0),
```
Comment (Sasha071201) [inline lib/common/components/building_component.dart:37]: I think you should just always set it to 100

File: lib/domain/services/economicy_service.dart [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import 'package:big_apple/data/services/game_state.dart';
+import 'package:big_apple/domain/entities/building_entity.dart';
+import 'package:big_apple/domain/entities/game_stat.dart';
+import 'package:big_apple/domain/services/building_service.dart';
+import 'package:flutter/material.dart';
+import 'package:rxdart/rxdart.dart';
+
+mixin EconomicyService implements BuildingService {
+  bool isEconomicServiceInitialized = false;
+  // ignore: unused_field
```
Comment (Sasha071201) [inline lib/domain/services/economicy_service.dart:16]: why is this here?

**Review by Sasha071201 — APPROVED** [review]

File: lib/big_apple_game.dart (hunk 1 above)
Comment (hrubalski) [reply lib/big_apple_game.dart]: fixed

File: lib/domain/services/economicy_service.dart (hunk 3 above)
Comment (hrubalski) [reply lib/domain/services/economicy_service.dart:16]: I'll remove it as soon as I create a dispose method for this service
